### PR TITLE
Add all signature stamps to GOBL envelope

### PR DESCRIPTION
**Pull request by samlown**

### CODE REVIEW

**Review by cavalle — APPROVED** [review]
LGTM!

File: lines.go [hunk 1]
```diff
@@ -46,9 +46,25 @@ func newConcepto(line *bill.Line, regime *tax.Regime) *Concepto {
...
+
+func lineSubjectToTax(line *bill.Line, r *tax.Regime) string {
+	if len(line.Taxes) == 0 {
+		return ObjetoImpNo
+	}
+	// Check if the line's VAT is exempt from VAT
+	for _, combo := range line.Taxes {
+		if combo.Category == tax.CategoryVAT {
+			rate := r.Rate(combo.Category, combo.Rate)
+			if rate != nil && rate.Exempt {
```
Comment (cavalle) [inline lines.go:64]: ⚠️ Watch out here ⚠️: In Mexico, invoice lines can be either non-subject or subject of the tax, and those that are subject can be exempt or not. The value `Exento` should be set in the key `TipoValor` if that happens. That means we need to provide a way to distinguish between the non-subject and the exempt cases.

**Review by cavalle — CHANGES_REQUESTED** [review]
See my comment

File: lines.go (hunk 1 above)
Comment (samlown) [reply lines.go:64]: right... so the best option here then would be to differentiate between lines that have taxes (subject) and those that don't (not-subject). I'll check the `TipoValor`, as there is also the case here of zero rated.

Comment (samlown) [reply lines.go]: Okay. So this is now fixed. The `TipoFactor` is now set appropriately. GOBL has also been updated to allow invoice lines without taxes, which implies the `ObjetoImp` will be set to `002`.

Comment (cavalle) [reply lines.go]: @samlown in other regimes, we use the "exempt" rate for both and then an extension specifies whether is an exemption or a non-subject case:

* Portugal: https://github.com/invopop/gobl/blob/main/regimes/pt/extensions.go#L46
* Italy: https://github.com/invopop/gobl/blob/main/regimes/it/extensions.go#L100

Comment (cavalle) [reply lines.go]: Previous comment updated to include TicketBAI

Comment (samlown) [reply lines.go]: yeah, they require explicit codes as they are more like VAT declarations and always require VAT to be present. But the case in Mexico is clear. The special code is explicit: "there are no taxes".

Comment (cavalle) [reply lines.go]: Thanks, that makes sense! I just thought that a person generating GOBL in different countries, unaware of the particularities of each local target format, may find surprising having to treat, seemingly, the same circumstance so differently depending on the country.

**Pull request merged**